framework/include/functions/PiecewiseLinearBase.h

```diff
@@ -28,12 +28,13 @@ class PiecewiseLinearBase : public PiecewiseBase
 
   PiecewiseLinearBase(const InputParameters & parameters);
 
-  virtual void initialSetup() override;
-  virtual Real value(Real t, const Point & pt) const override;
-  virtual Real timeDerivative(Real t, const Point & pt) const override;
-  virtual Real integral() const override;
-  virtual Real average() const override;
-  virtual void setData(const std::vector<Real> & x, const std::vector<Real> & y) override;
+  void initialSetup() override;
+  Real value(Real t, const Point & p) const override;
+  Real timeDerivative(Real t, const Point &) const override;
+  RealGradient gradient(Real, const Point & p) const override;
+  Real integral() const override;
+  Real average() const override;
+  void setData(const std::vector<Real> & x, const std::vector<Real> & y) override;
```

Comment:
```suggestion
  void initialSetup() final;
  Real value(Real t, const Point & p) const final;
  Real timeDerivative(Real t, const Point &) const final;
  RealGradient gradient(Real, const Point & p) const final;
  Real integral() const final;
  Real average() const final;
  void setData(const std::vector<Real> & x, const std::vector<Real> & y) final;
```

I suggest the best of both worlds per the previous discussion

Reply:
Huh?! Marking the methods final is a completely different issue. I'm not sure that's what we necessarily want. And I certainly don't want to incorporate such a design change into my PR. 

We can discuss whether we want to have a redundant `virtual` + `override` (per Zach) or go with the `override` only version (which I prefer).

Reply:
Sorry - I didn't follow then. I thought you were trying to take them away. I prefer just `override` in this case.

Reply:
Ok, right now we're using `virtual` + `override` ~3800 times and `override` without `virtual` ~1000 times in moose. This is not the hill I want to die on, so I'll just put the virtual back.

Reply:
Didn't see you comment because I was stomping my feet and reluctantly put the `virtual` back :-D

It's nit a big deal, but for the sake of consistency we should probably come to a decision what to do in the moose codebase. But I suggest doing that in a separate issue. By default _not_ making this change in this PR, where it is not strictly necessary is the right thing IMO.

Reply:
Hah - sounds good. Deferring to a later issue is the moose way, eh?